decisions/0009-schema-refactor.md

```diff
@@ -0,0 +1,95 @@
+---
+status: accepted 
+date: 2023-12-19
+deciders: Chris Mungall, Lee Ann 
+consulted: Anastasiya Prymolenna, Brynn Zalmanek, James Tessmer, Montana Smith, Sam Purvine, Yuri Corilo, Mark Miller, Michael Thorton, Alicia Clum, Mark Miller    
+informed: Emiley Eloe-Fadrosh, Shreyas Cholia, Eric Cavanna 
+---
+# Schema Refactoring: Monterey//Berkeley
+
+## Decisions Made
+
+### Class:PlannedProcess
+* `Class:PlannedProcess` will be created as a root class 
+* All "Activity" classes are now of `Class:PlannedProcess`, and the word "Activity" will be removed
+
+### Class:MaterialProcessing
+* `Class:MaterialProcessing` will be created and be a subclass of `Class:PlannedProcess`
+* All processes involving material to material transformations (Biosamples to ProcessedSample, ProcessedSample to ProcessedSample) will be subclasses of `Class:MaterialProcessing` (examples: `Class:Extraction`, `Pooling`, `MixingProcess`, etc.). 
+* The output of any subclasses of `Class:MaterialProcessing` will have an output of ProcessedSample
+* `Class:ChemicalConversionProcess` will replace `Class:ReactionProcess`
+* `Class:BiosampleProcessing` will be removed. Subclasses of `Class:MaterialProcessing` will replace what was captured. 
+* Linking a protocol to a `Class:MaterialProcess` subclass can happen at the individual class level, or at the aggregation of processes via the `Class:ProtocolExecution`
+* `slot:catalyzed_by` is instantiated on `Class:ChemicalConversionProcess` with `Range:Class:Solution`.
+  * `Class:Solution` has `slot:has_solution_components` with `Range:Class:SolutionComponent` where `slot:compound` has `Range:Class:ProteolyticEnzymeEnum`.
```

Comment:
Reading this bullet point, it's not clear to me whether `slot:compound` is a slot on `Class:Solution` or a slot on `Class:SolutionComponent`.

Reply:
> it's not clear to me whether `slot:compound` is a slot on `Class:Solution` or a slot on `Class:SolutionComponent`.

The version in the [latest commit](https://github.com/microbiomedata/NMDC_documentation/pull/36/files#diff-5c094c1fa6007fb35653d174ce0f36cb3652d4abe7ae07f4455deaafa7ec884aR24) is no longer ambiguous to me. I consider this issue resolved.

Reply:
This is the definition of compound in the Berkeley Schema
```
compound:
    domain: SolutionComponent
    any_of:
      - range: CompoundEnum
      - range: ProteolyticEnzymeEnum
``` 
[Here in the schema](https://github.com/microbiomedata/berkeley-schema-fy24/blob/9baac0603c17cf038f93bcfcef5bd250d26af138/src/schema/nmdc.yaml#L3591C2-L3591C2)

What I think is happening is the same issue we had to do a work around for where using 'any_of' doesn't use the listed ranges and defaults to having the range of string.

Reply:
We could do something like [this](https://github.com/microbiomedata/berkeley-schema-fy24/blob/9baac0603c17cf038f93bcfcef5bd250d26af138/src/schema/nmdc.yaml#L1476C6-L1476C6) which seems to work for the DOI enums

Reply:
issue created to fix this https://github.com/microbiomedata/nmdc-schema/issues/1666